tests/TestTCP.hs

```diff
@@ -647,9 +647,10 @@ testReconnect = do
     -- The first time we close the socket before accepting the logical connection
     count <- modifyMVar counter $ \i -> return (i + 1, i)
 
-    -- The client is responsible for taking this MVar before trying to connect
-    -- again.
-    putMVar socketClosed (WaitSocketFree socketFree)
+    -- Wait 100ms after the socket closes, to (hopefully) ensure that the client
+    -- knows the connection is closed, and sending on that socket will therefore
+    -- fail.
+    putMVar socketClosed (WaitSocketFree (socketFree >> threadDelay 100000))
```

Comment:
Rather than inserting a delay, couldn't the client send messages repeatedly until the failure is detected?

I'm trying to see if we can arrange the test to be time insensitive.